pandas/tests/groupby/test_groupby.py

```diff
@@ -1951,3 +1951,13 @@ def test_groupby_only_none_group():
     expected = pd.Series([np.nan], name="x")
 
     tm.assert_series_equal(actual, expected)
+
+
+def test_groupby_duplicate_index():
+    # GH#29189 the groupby call here used to raise
+    ser = pd.Series([2, 5, 6, 8], index=[2.0, 4.0, 4.0, 5.0])
```

Comment:
Do we not already have a similar test elsewhere? Not a blocker for this PR but feels like we should have a parametrized or more broad-reaching test for this in groupby

Reply:
That's a good question and I dont have a good answer.  i'll take a look tomorrow; most likely parametrizing/de-duplicating will need a dedicated pass

Reply:
It looks like this affects the exceptions we get for other Index subclasses too, so I'll prioritize fleshing this out in the AM